spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/MapTaskExecutionDao.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.task.repository.dao;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.cloud.task.repository.TaskExecution;
+
+/**
+ * Stores Task Execution Information to a in memory map.
+ *
+ * @author Glenn Renfro
+ */
+public class MapTaskExecutionDao implements TaskExecutionDao {
+
+	private Map<String, TaskExecution> taskExecutions;
+
+	public MapTaskExecutionDao() {
+		taskExecutions = new HashMap<String, TaskExecution>();
+	}
+
+	@Override
+	public void saveTaskExecution(TaskExecution taskExecution) {
+		taskExecutions.put(taskExecution.getExecutionId(), taskExecution);
+	}
+
+	@Override
+	public void updateTaskExecution(TaskExecution taskExecution) {
+		taskExecutions.put(taskExecution.getExecutionId(), taskExecution);
+	}
+
+	public Map<String, TaskExecution> getTaskExecutions(){
+		return taskExecutions;
```

Comment:
This exposes the underlying data structure which means someone could modify it without using the dao.  This needs to return an unmodifiable copy...